Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md

```diff
@@ -238,3 +238,38 @@ parameters:
 ```
 
 * PVCs created using the new storageclass will be encrypted.
+
+
+## Enable Read affinity for RBD volumes
+
+Ceph CSI supports mapping RBD volumes with krbd options to allow
+serving reads from an OSD in proximity to the client, according to
+OSD locations defined in the CRUSH map. The format of the krbd option
+is: `"read_from_replica=localize,crush_location=type1:value1|type2:value2"`.
+
+Refer to the [krbd-options](https://docs.ceph.com/en/latest/man/8/rbd/#kernel-rbd-krbd-options)
+for more details.
+
+Execute the following steps:
+
+* Add labels to Kubernetes nodes following procedure similar to how it is mentioned in
+[osd-topology](../../CRDs/Cluster/ceph-cluster-crd.md#osd-topology).
+
+* (optional) Rook will pass the labels mentioned in [osd-topology](../../CRDs/Cluster/ceph-cluster-crd.md#osd-topology)
+as the default set of labels. This can overridden to supply custom labels using the following command.
+```console
+kubectl patch cm rook-ceph-operator-config -nrook-ceph -p $'data:\n "CSI_CRUSH_LOCATION_LABELS": "<custom-labels>"'
+```
+
+Ceph CSI will extract the crush location from the topology labels found on the node
+and pass it though krbd options during mapping RBD volumes.
+
+Example:
+`"--options read_from_replica=localize,crush_location=zone:east-zone1|region:east"`
+
+This feature is enabled by default. To disable it,
```

Comment:
Do we need to worry about the ceph version or kernel supporting it (back to Pacific) before we enable it by default? Or the topology will be ignored where not supported?

Reply:
https://docs.ceph.com/en/latest/man/8/rbd/#kernel-rbd-krbd-options

This document tells there's support from  5.8 and
This mode is safe for general use only since Octopus (i.e. after “ceph osd require-osd-release octopus”). Otherwise it should be limited to read-only workloads such as images mapped read-only everywhere or snapshots.